test/framework/src/test/java/org/elasticsearch/test/disruption/NetworkDisruptionIT.java

```diff
@@ -102,4 +112,56 @@ public void testNetworkPartitionRemovalRestoresConnections() throws Exception {
         }
     }
 
+    public void testTransportRespondsEventually() throws InterruptedException {
+        internalCluster().setBootstrapMasterNodeIndex(0);
+        internalCluster().ensureAtLeastNumDataNodes(randomIntBetween(3, 5));
+        final NetworkDisruption.DisruptedLinks disruptedLinks;
+        if (randomBoolean()) {
+            disruptedLinks = TwoPartitions.random(random(), internalCluster().getNodeNames());
+        } else {
+            disruptedLinks = NetworkDisruption.Bridge.random(random(), internalCluster().getNodeNames());
+        }
+
+        NetworkDisruption networkDisruption = new NetworkDisruption(disruptedLinks, randomFrom(new NetworkDisruption.NetworkUnresponsive(),
+            new NetworkDisruption.NetworkDisconnect(), NetworkDisruption.NetworkDelay.random(random())));
+        internalCluster().setDisruptionScheme(networkDisruption);
+
+        networkDisruption.startDisrupting();
+
+        CountDownLatch latch = new CountDownLatch(100);
+        for (int i = 0; i < 100; ++i) {
+            TransportService transportService = internalCluster().getInstance(TransportService.class);
+            TransportService target = internalCluster().getInstance(TransportService.class);
+
+            transportService.sendRequest(target.getLocalNode(), ClusterHealthAction.NAME, new ClusterHealthRequest(),
+                new TransportResponseHandler<>() {
+                @Override
+                public void handleResponse(TransportResponse response) {
+                    latch.countDown();
```

Comment:
I like to clarify that we're not double-counting anything when using a `CountDownLatch`. How about something like this for an `AtomicBoolean responded`:

```suggestion
                    assertTrue(responded.compareAndSet(false, true));
                    latch.countDown();
```

Reply:
Fixed in 3a4da10ac241a395ded339244b16d079aa3abf70